ui/org.eclipse.pde.launching/src/org/eclipse/pde/launching/AbstractPDELaunchConfiguration.java

```diff
@@ -193,26 +195,31 @@ private String[] updateVMArgumentWithAdditionalArguments(String[] args, boolean
 		return args;
 	}
 
-	private static final String SEPARATOR = "."; //$NON-NLS-1$
 
-	private boolean isEclipseBuild424OrMore(String buildId) {
-		int major = 0, minor = 0;
-		try {
-			StringTokenizer st = new StringTokenizer(buildId, SEPARATOR, true);
-			major = Integer.parseInt(st.nextToken()); // minor
-
-			if (st.hasMoreTokens()) { // minor
-				st.nextToken(); // consume delimiter
-				minor = Integer.parseInt(st.nextToken());
-			}
-			if (major >= 4 && minor >= 24) {
-				return true;
+	private boolean isEclipseBundleGreaterThanVersion(int major, int minor) {
+		Version version;
+		String product = "org.eclipse.platform.ide"; //$NON-NLS-1$
+		IExtensionRegistry registry = Platform.getExtensionRegistry();
+		IExtensionPoint point = registry.getExtensionPoint("org.eclipse.core.runtime.products"); //$NON-NLS-1$
```

Comment:
Isn't that also for *current* instance, and not for the one in the target platform? I have no IDE right now, but PDE launch must know the target platform content and versions of the bundles there.

Reply:
I added a target platform, and when creating a new Eclipse Configuration in the launched eclipse, it provides the "target" build id.  I need to test for the current workspace after  installing it as feature to the base eclipse.

Reply:
you can check the target version with `PluginRegistry.findModel(IPDEBuildConstants.BUNDLE_OSGI)`

Reply:
The 3.8 constant was added 10 years ago, not in 4.19. You can either add a new TARGET constant, like documented in `ICoreConstants.TARGET_VERSION_LATEST`, or do the `findModel` check yourself in `AbstractPDELaunchConfiguration`.